expression/aggregation/concat.go

```diff
@@ -88,7 +95,17 @@ func (cf *concatFunction) Update(evalCtx *AggEvaluateContext, sc *stmtctx.Statem
 	for _, val := range datumBuf {
 		cf.writeValue(evalCtx, val)
 	}
-	// TODO: if total length is greater than global var group_concat_max_len, truncate it.
+	if cf.maxLen > 0 && uint64(evalCtx.Buffer.Len()) > cf.maxLen {
+		i := mathutil.MaxInt
+		if uint64(i) > cf.maxLen {
+			i = int(cf.maxLen)
+		}
+		evalCtx.Buffer.Truncate(i)
```

Comment:
`Buffer.Len()` returns a `int` typed value, I'm curious about what will happen when we append more than 4GB bytes into that buffer.:blush: 

The interesting part is, MySQL limits `group_concat_max_len` to the max uint64 integer in 64-bit systems, but we use a buffer which seems only capable of 4GB bytes.:smirk:

Reply:
That's a problem. Since it's not likely that this limit is reached for now, I don't make effort modify the implementation of `Buffer`. Maybe we should make a note here ? -.-